samples/nrf9160/modem_shell/src/ltelc/ltelc_api.c

```diff
@@ -0,0 +1,944 @@
+	while ((tmp_ptr = strstr(tmp_ptr, AT_CMD_PDP_CONTEXT_READ_RSP_DELIM)) !=
+	       NULL) {
+		++tmp_ptr;
+		++lines;
+	}
+
+	/* TODO: DBG logging for:
+	 * printf("Device contains %d lines of DNS info for CID=%d\n", lines, populated_info->cid);
+	 */
+
+	/* Parse the response: */
+	{
+		ret = at_params_list_init(
+			&param_list, AT_CMD_PDP_CONTEXT_READ_INFO_PARAM_COUNT);
+		if (ret) {
+			printk("Could not init AT params list, error: %d\n",
+			       ret);
+			return ret;
+		}
+
+		/* TODO: Make this a while loop */
+parse:
+		resp_continues = false;
+		ret = at_parser_max_params_from_str(at_ptr, &next_param_str,
+						    &param_list, 13);
+		if (ret == -EAGAIN) {
+			resp_continues = true;
+		} else if (ret == -E2BIG) {
+			printk("E2BIG, error: %d\n", ret);
+		} else if (ret != 0) {
+			printk("Could not parse AT response for %s, error: %d\n",
+			       at_cmd_pdp_context_read_info_cmd_str, ret);
+			goto clean_exit;
+		}
+
+		/* Read primary DNS address */
+		param_str_len = sizeof(dns_addr_str);
+		ret = at_params_string_get(
+			&param_list,
+			AT_CMD_PDP_CONTEXT_READ_INFO_DNS_ADDR_PRIMARY_INDEX,
+			dns_addr_str, &param_str_len);
+		if (ret) {
+			printk("Could not parse dns str for cid %d, err: %d\n",
+			       populated_info->cid, ret);
+			goto clean_exit;
+		}
+		dns_addr_str[param_str_len] = '\0';
+
+		if (dns_addr_str != NULL) {
+			int family = net_utils_sa_family_from_ip_string(
+				dns_addr_str);
+			if (family == AF_INET) {
+				struct in_addr *addr =
+					&(populated_info->dns_addr4_primary);
+				(void)inet_pton(AF_INET, dns_addr_str, addr);
+			} else if (family == AF_INET6) {
+				struct in6_addr *addr6 =
+					&(populated_info->dns_addr6_primary);
+				(void)inet_pton(AF_INET6, dns_addr_str, addr6);
+			}
+		}
+
+		/* Read secondary DNS address */
+		param_str_len = sizeof(dns_addr_str);
+
+		ret = at_params_string_get(
+			&param_list,
+			AT_CMD_PDP_CONTEXT_READ_INFO_DNS_ADDR_SECONDARY_INDEX,
+			dns_addr_str, &param_str_len);
+		if (ret) {
+			printk("Could not parse dns str, err: %d\n", ret);
+			goto clean_exit;
+		}
+		dns_addr_str[param_str_len] = '\0';
+
+		if (dns_addr_str != NULL) {
+			int family = net_utils_sa_family_from_ip_string(
+				dns_addr_str);
+			if (family == AF_INET) {
+				struct in_addr *addr =
+					&(populated_info->dns_addr4_secondary);
+				(void)inet_pton(AF_INET, dns_addr_str, addr);
+			} else if (family == AF_INET6) {
+				struct in6_addr *addr6 =
+					&(populated_info->dns_addr6_secondary);
+				(void)inet_pton(AF_INET6, dns_addr_str, addr6);
+			}
+		}
+
+		/* Read link MTU if exists: */
+		ret = at_params_int_get(&param_list,
+					AT_CMD_PDP_CONTEXT_READ_INFO_MTU_INDEX,
+					&(populated_info->mtu));
+		if (ret) {
+			/* Don't care if it fails: */
+			ret = 0;
+			populated_info->mtu = 0;
+		}
+
+		if (resp_continues) {
+			at_ptr = next_param_str;
+			iterator++;
+			if (iterator >= lines) {
+				/* Should not happen, just in case... TODO: add assert?*/
+				ret = -666;
```

Comment:
Perhaps just use some errno value? And assert does sound fine here.

Reply:
Changed to assert